src/devtools/useAtomDevtools.ts

```diff
@@ -59,16 +72,36 @@ export function useAtomDevtools<Value>(
 
   useEffect(() => {
     if (extension) {
+      const setValueIfWritable = (value: Value) => {
+        if (typeof setValue === 'function') {
+          ;(setValue as SetAtom<Value, void>)(value)
+          return
+        }
+        console.warn(
+          '[Warn] you cannot do write operations (Time-travelling, etc) in read-only atoms\n',
+          anAtom
+        )
+      }
       devtools.current = extension.connect({ name: atomName })
       const unsubscribe = devtools.current.subscribe((message: Message) => {
-        if (message.type === 'DISPATCH' && message.state) {
+        if (message.type === 'ACTION' && message.payload) {
+          try {
+            setValueIfWritable(JSON.parse(message.payload))
+          } catch (e) {
+            console.error(
+              'please dispatch a serializable value that JSON.parse() support\n',
+              e
+            )
+          }
+        } else if (message.type === 'DISPATCH' && message.state) {
           if (
             message.payload?.type === 'JUMP_TO_ACTION' ||
             message.payload?.type === 'JUMP_TO_STATE'
           ) {
             isTimeTraveling.current = true
+
+            setValueIfWritable(JSON.parse(message.state))
           }
-          setValue(JSON.parse(message.state))
```

Comment:
we don't need to set value if the type is not JUMP_TO_ACTION/STATE?

Reply:
In other libs, we don't do that tbh, and they work well! Do we need it?

Reply:
Oh, I see. So, it was kind of mistake only in jotai. No, I was just curious about the change.